osh/hub/scripts/osh-worker-manager

```diff
@@ -0,0 +1,220 @@
+
+    # Check if a worker with this name already exists
+    try:
+        kobo_worker = models.Worker.objects.get(name=worker_name)
+    except ObjectDoesNotExist:
+        # If not, create a new one
+        kobo_worker = models.Worker.create_worker(worker_name)
+
+    # TODO: Detect worker architecture automatically.
+    try:
+        kobo_worker.arches.add(models.Arch.objects.get(name="x86_64"))
+        kobo_worker.arches.add(models.Arch.objects.get(name="noarch"))
+    except ObjectDoesNotExist:
+        print("x86_64 or noarch do not exist in Arch model.", file=sys.stderr)
+        kobo_worker.delete()
+        sys.exit(1)
+
+    try:
+        kobo_worker.channels.add(models.Channel.objects.get(name="default"))
+    except ObjectDoesNotExist:
+        print("Default channel does not exist.", file=sys.stderr)
+        kobo_worker.delete()
+        sys.exit(1)
+
+    # Each single use virtual machine allocated should get exactly one task
+    # and it should be deleted after that.
+    kobo_worker.max_load = 1
+
+    kobo_worker.save()
+    print(f"Created kobo worker for hostname {worker_name}")
+    return kobo_worker
+
+
+def start_worker(kobo_worker):
+    """
+    This function works in below steps:
+    1. Generate a `worker.conf` with the newly generated `worker_key`.
+    2. Copy the `worker.conf` to the worker.
+    3. Move the file to `/etc/osh/worker.conf` and change it's permissions to `root:root`.
+    4. Start the `osh-worker` service on the virtual machine.
+    """
+    # This function looks like a hack and we should investigate if it could be done better.
+    # For example, through ansible or paramiko.
+
+    from django.conf import settings
+    from django.template.loader import render_to_string
+
+    # Generate /etc/osh/worker.conf through jinja template
+    worker_conf = render_to_string("worker.conf.j2",
+                                   {"SINGLE_USE_WORKER_OSH_HUB_URL": settings.SINGLE_USE_WORKER_OSH_HUB_URL,
+                                    "OSH_WORKER_KEY": kobo_worker.worker_key})
+
+    # Create a temporary file and save the template
+    # Set the parameter to `delete_on_close=False` on python3.12 update
+    # https://docs.python.org/3.12/whatsnew/3.12.html#tempfile
+    # with tempfile.NamedTemporaryFile(delete=False) as temporary_file:
+    try:
+        temporary_file = tempfile.NamedTemporaryFile()
+        temporary_file.write(worker_conf.strip().encode())
+        # TODO: File is closed, but not removed if `delete_on_close` is True.
+        # temporary_file.close()
+        temporary_file.flush()
+        # and transfer it to new worker.
+        # Copy worker.conf to ec2-user user's home directory
+        # This requires the hub to have ssh access to the new worker
+        scp_cmd = f"scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=20 -i {settings.SINGLE_USE_WORKER_SSH_PRIVATE_KEY} {temporary_file.name} ec2-user@{kobo_worker.name}:worker.conf"
```

Comment:
The username for ssh should be configurable through the settings. It is not always same for different virtual machines.